cli/src/lib.rs

```diff
@@ -513,35 +513,44 @@ fn override_toolchain(cfg_override: &ConfigOverride) -> Result<RestoreToolchainC
             Ok(version)
         }
 
+        fn is_solana_version_installed(version: &str) -> Result<bool> {
+            let output = std::process::Command::new("solana").arg("--version").output()?;
+            let output_version = std::str::from_utf8(&output.stdout)?;
+            Ok(output_version.contains(version))
+        }
```

Comment:
There is a difference between an installed version and the current version — this function only checks whether the given version is the current version.